1-js/06-advanced-functions/09-call-apply-decorators/02-delay/solution.md

```diff
@@ -11,22 +11,22 @@ function delay(f, ms) {
 
 let f1000 = delay(alert, 1000);
 
-f1000("test"); // shows "test" after 1000ms
+f1000("teste"); // exibe "teste" após 1000ms
 ```
 
-Please note how an arrow function is used here. As we know, arrow functions do not have own `this` and `arguments`, so `f.apply(this, arguments)` takes `this` and `arguments` from the wrapper.
+Nota como é utilizada aqui uma função de seta (ou função anónima). Como sabemos, as funções de seta não possuem `this` e `arguments` próprios, então `f.apply(this, arguments)` recebe o `this` e `arguments` da função envolvente.
```

Comment:
```suggestion
Note como é utilizada aqui uma função de seta. Como sabemos, as funções de seta não possuem `this` e `arguments` próprios, então `f.apply(this, arguments)` recebe o `this` e `arguments` da função envolvente.
```

Funções de seta e funções anônimas não são conceitos equivalentes, funções de seta podem ser nomeadas e funções anônimas podem ser declaradas usando `function`

Reply:
@Peruibeloko, I think you are not remembering that anonymous functions could be named.

To illustrate what I want to tell, consider this humble example below:

```js
let getName = function(name) { return name }
console.log(`${getName('Marcos')} was called by ${getName.name} function`)

getName = (name) => name
console.log(`${getName('Marcos')} was called by ${getName.name} function`)
```

I recommend you to try to run the code above to see what I am talking about.

Reply:
@nazarepiedady On your example, neither function is anonymous:

![image](https://github.com/javascript-tutorial/pt.javascript.info/assets/3859329/94e5490d-4559-4ffe-a52e-27a97ce5896e)

- The first one creates an anonymous function, which then gets named as `getName1`
- The second one creates a named arrow function called `getName2`.

An anonymous function is, [by definition](https://developer.mozilla.org/en-US/docs/Glossary/Function#different_types_of_functions), a function without a name.

A better example would be:

```js
// Anonymous function expression
console.log((function() {}).name)

// Named function declaration
console.log((function namedDeclaration() {}).name)

// Anonymous arrow function
console.log((() => {}).name)

// Named arrow function
const namedArrow = () => {}
console.log(namedArrow.name)
```

Which correctly prints out:

![image](https://github.com/javascript-tutorial/pt.javascript.info/assets/3859329/dc18f9e7-359c-4654-b9e9-22ea519bea6b)

Reply:
@Peruibeloko, You are absolutely correct, I was not remembered about this approach.